### DISCUSSION TITLE
OpenAPI generation failed with internal enums and with pydantic >=1.6.*

### DISCUSSION THREAD

**Discussion by arxell** [fastapi/fastapi#8973] [Questions] [answered]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.

### Example

```Python
from fastapi import FastAPI
from pydantic import BaseModel
from enum import Enum

app = FastAPI()


class EnumExample(Enum):
    a = 1


class Foo(BaseModel):
    class EnumExample(Enum):
        a = EnumExample.a
        b = 2

    x: EnumExample


class Bar(BaseModel):
    class EnumExample(Enum):
        a = EnumExample.a
        b = 2

    x: EnumExample


class TestResp(BaseModel):
    foo: Foo
    bar: Bar


@app.get("/", response_model=TestResp)
async def test():
    return TestResp(foo=Foo(x=Foo.EnumExample.a), bar=Bar(x=Bar.EnumExample.b))

```

### Description

* Open the browser and call the endpoint `http://127.0.0.1:8000/docs`.
* It returns an error `Internal Server Error /openapi.json
`.
* The problem is relevant only with pydantic>=1.6.* versions

### StackTrace
```
  File "/Users/arxell/repo/mdcm/alfa/venv39/lib/python3.9/site-packages/fastapi/applications.py", line 143, in openapi
    return JSONResponse(self.openapi())
  File "/Users/arxell/repo/mdcm/alfa/venv39/lib/python3.9/site-packages/fastapi/applications.py", line 121, in openapi
    self.openapi_schema = get_openapi(
  File "/Users/arxell/repo/mdcm/alfa/venv39/lib/python3.9/site-packages/fastapi/openapi/utils.py", line 348, in get_openapi
    definitions = get_model_definitions(
  File "/Users/arxell/repo/mdcm/alfa/venv39/lib/python3.9/site-packages/fastapi/utils.py", line 31, in get_model_definitions
    model_name = model_name_map[model]
KeyError: <enum 'EnumExample'>
```

### Environment

* OS: macOS
* FastAPI: 0.62.0
* Python: 3.9.0
* pydantic: 1.7.3

**Comment by includeamin** [comment]
#2061

**Comment by includeamin** [comment]
#1986

**Comment by brumar** [comment, marked as answer]
We had the same problem, updating to fastapi to 0.65 worked for us.

**Comment by Sailokeshg** [comment]
> We had the same problem, updating to fastapi to 0.65 worked for us.

How do i update existing version of fastapi to latest in ubuntu. I'm getting some errors when using pip install fastapi

**Comment by JarroVGIT** [comment]
Either uninstall it first, or use `pip install fastapi --upgrade`

**Comment by Sailokeshg** [comment]
That worked  Successfully installed fastapi-0.78.0 starlette-0.19.1 , but in http://127.0.0.1:8000/openapi.json . it is still showing version of fastapi as "0.1.0"

**Comment by JarroVGIT** [comment]
That is not the version of FastAPI, but the default version of your API itself. You can set the version like so:

`app = FastAPI(version='15.0.3')`

When not setting a version when initiating a `FastAPI()` application, it defaults to `0.1.0` which you will see in the docs. See also: https://fastapi.tiangolo.com/tutorial/metadata/?h=version#metadata-for-api
